src/core/execute.c

```diff
@@ -4878,6 +4885,17 @@ int exec_context_get_clean_mask(ExecContext *c, ExecCleanMask *ret) {
         return 0;
 }
 
+int exec_context_cpu_affinity_from_numa(ExecContext *c) {
+        assert(c);
+
+        if (!c->numa_policy.derive_cpu_affinity || !c->numa_policy.nodes.set)
```

Comment:
Maybe add a comment here that this finalizes the configuration, and since we don't expect the numa list to change, we modify the original structure?

Reply:
Should there be a warning if CPUAffinity=numa is set but it has no effect because the numa mask is not set?

Reply:
> Maybe add a comment here that this finalizes the configuration, and since we don't expect the numa list to change, we modify the original structure?

I can add the comment but it would be slightly misleading as this code is running after `fork()` in a child so actually modifying the `ExecContext` here doesn't modify the original struct in the parent. IIUC @poettering explicitly wanted this late evaluation.